runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/config/GrpcBindingConfig.java

```diff
@@ -77,7 +76,7 @@ public final class GrpcBindingConfig
     private final HttpGrpcHeaderHelper helper;
     private final Long2ObjectHashMap<CatalogHandler> handlersById;
     private final Int2ObjectHashMap<GrpcProtobufConfig> protobufsBySchemaId;
-    private final ObjectHashSet<CatalogedConfig> catalogs;
+    private final List<CatalogedConfig> catalogs;
```

Comment:
I don't think we need both `List<CatalogedConfig> catalogs` and `Long2ObjectHashMap<CatalogHandler> handlersById`.

Seems like we need a data structure that has:
```
final CatalogHandler handler;
final String subject;
final String version;
int schemaId;
GrpcProtobufConfig protobuf;
```
and a list of these, flattening them for each catalog + schema combination, using same catalog handler for schemas on each catalog.

Then in resolveProtobufs, we just need to iterate this list, checking to see if schemaId changes compared to most recent one, and update it and newly parsed protobuf if necessary.

Then resolveMethod can stream over this list instead, mapping to GrpcProtobufConfig, before doing the rest.